src/thor/costmatrix.cc

```diff
@@ -766,42 +768,48 @@ void CostMatrix::CheckForwardConnections(const uint32_t source,
     // If we came down here, we know this opposing edge is either settled, or it's a
     // target correlated edge which hasn't been pulled out of the queue yet, so a path
     // has been found to the end node of this directed edge
-    const auto& opp_edgestate = edgestatus_[MATRIX_REV][target];
-    EdgeStatusInfo oppedgestatus = opp_edgestate.Get(oppedge);
-    const auto& opp_edgelabels = edgelabel_[MATRIX_REV][target];
-    uint32_t opp_predidx = opp_edgelabels[oppedgestatus.index()].predecessor();
-    const BDEdgeLabel& opp_el = opp_edgelabels[oppedgestatus.index()];
+    const auto& rev_edgestate = edgestatus_[MATRIX_REV][target];
+    EdgeStatusInfo rev_edgestatus = rev_edgestate.Get(rev_edgeid);
+    const auto& rev_edgelabels = edgelabel_[MATRIX_REV][target];
+    uint32_t rev_predidx = rev_edgelabels[rev_edgestatus.index()].predecessor();
+    const BDEdgeLabel& rev_label = rev_edgelabels[rev_edgestatus.index()];
 
     // Special case - common edge for source and target are both initial edges
-    if (pred.predecessor() == kInvalidLabel && opp_predidx == kInvalidLabel) {
-      // TODO: shouldnt this use seconds? why is this using cost!?
-      float s = std::abs(pred.cost().secs + opp_el.cost().secs - opp_el.transition_cost().cost);
+    if (fwd_pred.predecessor() == kInvalidLabel && rev_predidx == kInvalidLabel) {
+      // bail if either edge wasn't allowed (see notes in SetSources/Targets)
+      if (!fwd_pred.path_id() || !rev_label.path_id()) {
+        return;
+      }
```

Comment:
this is the actual change: `path_id()` is not used in CostMatrix of course, so I use it in `SetSources/Targets` to record whether the edge we want to connect on is actually traversable in both directions.